GitCommands/Git/GitCreateTagCmd.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GitCommands.Git
+{
+    public enum TagOperation
+    {
+        Lightweight = 0,
+        Annotate,
+        SignWithDefaultKey,
+        SignWithSpecificKey
+    };
+
+    public static class TagOperationExtensions
+    {
+        public static bool CanProvideMessage(this TagOperation operationType)
+        {
+            switch (operationType)
+            {
+                case TagOperation.Lightweight:
+                    return false;
+                case TagOperation.Annotate:
+                case TagOperation.SignWithDefaultKey:
+                case TagOperation.SignWithSpecificKey:
+                    return true;
+                default:
+                    throw new NotSupportedException("Invalid TagOperation: " + operationType);
+            }
+        }
+    }
+
+    public sealed class GitCreateTagArgs
```

Comment:
Its purpose is to group the arguments, which I don't want to pass through the constructor.

Reply:
Yes, I understand the purpose, I think it is the right thing to do.
However these are *input* parameters, meaning that they should not change once inside the target method. If they do - they could cause side effects and inconsistent state.
For the same reason I asked to convert public fields to properties - the latter are an encapsulation with added benefits (readonly, validation etc)

Reply:
Because these are input parameters, I create the object, pass it to a method and forget about it, not reusing it. If I were to make it immutable, I would have to either provide a constructor with each field going through it (which I didn't want to do in the first place) or provide some kind of builder which is a bit of overkill in my opinion. All I want is to group args together, fill them one by one like here https://github.com/gitextensions/gitextensions/blob/eab1aab4ee057177a84311e554950eb207684a18/GitUI/CommandsDialogs/FormCheckoutBranch.cs#L263 and pass to a method. I am aware of possibility of side effects and inconsistent state but I see no risk here.